src/wwwroot/app/services/httpService.js

```diff
@@ -1,37 +1,44 @@
 const BASE_URL = 'http://localhost:53031/api/';
 
 export default class HttpService {
-   constructor($http) {
+   constructor($http, $q) {
       'ngInject';
       this.http = $http;
+      this.$q = $q;
```

Comment:
really, I don't like to adding new props into `this`.
You can try add new variable into line number 2, and init it in yours constructor.
Please, think about this.